File: fishtank/static-object.cpp
#include "AquariumController.h"
#include <boost/bind.hpp>
#include <framework/debug.hpp>
#include <framework/resource.hpp>
#include <fstream>
#include "main.hpp"
#include "static-object.hpp"
#include <string>

using namespace boost;
using namespace std;

StaticObject::StaticObject(boost::shared_ptr<const Model> model, const std::string& propertiesFile, const Eigen::Vector3f& position) :
	Object(model, position)
{
	LoadProperties(propertiesFile);
}

void StaticObject::LoadProperties(const string& propertiesFile)
{
	string s;
	ifstream input_file((datadir + "/Objecten/" + propertiesFile + ".oif").c_str());

	//scaling
	getline(input_file, s);
	float object_height = atof(s.c_str());

	getline(input_file, s);/// randomize
	float n = atof(s.c_str());
	object_height = object_height + my_random() * n;

	if(model){
		scale = object_height / (model->bb_h.y() - model->bb_l.y());
	}else{
		scale = object_height;
		debug(LOG_ERROR) << "Trying to load properties for object that has no model, cant determine scaling";
	}
}

File: fishtank/Settings/static-object.lua
class 'StaticObject' (Object)

do
	local model_matrix = Eigen.Matrix4f.create({
		-1, 0, 0, 0,
		 0, 0, 1, 0,
		 0, 1, 0, 0,
		 0, 0, 0, 1
	})

	function StaticObject:__init(model_name, object_height, pos)
		local model = model_name
		if type(model) == 'string' then
			model = loadModel('Objecten/Modellen', model, model_matrix)
		end

		Object.__init(self, model, pos)
		self.scale = object_height / (model.bb_h.y - model.bb_l.y)
		self.pos = pos
	end
end
